### PR TITLE
Add purger functionality

### PR DESCRIPTION
**Pull request by FreTimmerman**
closes https://github.com/nerdland-unofficial-fans/nerdlandbot_js/issues/2

### CODE REVIEW

**Review by Eva774 — CHANGES_REQUESTED** [review]
I would especially like the channel to be added as an (optional) argument before merging. My comments about the description and the deletion of the bot's own messages are more optional remarks :)

File: commands/purger.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+const addNewPurger = async interaction => {
+  if (!await verifyAdminAsync(interaction)) { return }
+
+  const description = interaction.options.getString('description')
+  const cronTime = interaction.options.getString('cron_time') ?? '0 0 * * *'
+  const maxAge = interaction.options.getInteger('max_age')
+  const channel = interaction.channel
+  const guild = await getGuild(interaction.guildId)
+
+  if (channel.isVoiceBased) {
```
Comment (Eva774) [inline commands/purger.js:20]: The bot should also check whether it is allowed to remove messages in the requested channel

File: tasks/purgeChannel.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+          lastFetchedMessageId = fetchedMessages.last().id
+
+          // filter for bulk removal
+          const messagesToBulkDelete = fetchedMessages.filter(message => checkBulk(message, purger.maxAge))
+
+          // add non bulk removable message to messagesToDelete
+          messagesToDeleteIndividually = messagesToDeleteIndividually.concat(fetchedMessages.filter(message => checkNonBulk(message, purger.maxAge)))
+
+          // actually delete them
+          const deletedMessages = await channelToPurge.bulkDelete(messagesToBulkDelete)
```
Comment (woutermeuwis) [inline tasks/purgeChannel.js:88]: Does the bulk delete also limit the max amount of messages like the fetch does?
If not, wouldn't it be more efficient to delete them all together after the loop?

File: tasks/purgeChannel.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+const { Collection } = require('discord.js')
+
+const tasks = {}
+let _client
+
+const limit = 100 // 100 is the discord API limit SADFACE.JPG
+
+const twoWeeksInMilliSeconds = 14 * 24 * 60 * 60 * 1000
+
+const checkIndividual = (message, maxAge) => {
```
Comment (woutermeuwis) [inline tasks/purgeChannel.js:14]: not a big fan of these names as they indicate how they are used but not what they do.
Would prefer something more like 

shouldDeleteMessage
canDeleteInBulk
shouldDeleteIndividually

File: tasks/purgeChannel.js (hunk 2 above)
Comment (FreTimmerman) [reply tasks/purgeChannel.js:88]: i believe it also has the limit

**Review by Eva774 — APPROVED** [review]

**Pull request merged**